source/base/polynomials_bdm.cc

```diff
@@ -27,7 +28,8 @@ DEAL_II_NAMESPACE_OPEN
 
 template <int dim>
 PolynomialsBDM<dim>::PolynomialsBDM(const unsigned int k)
-  : polynomial_space(Polynomials::Legendre::generate_complete_basis(k))
+  : TensorPolynomialsBase<dim>(k, compute_n_pols(k))
+  , polynomial_space(Polynomials::Legendre::generate_complete_basis(k))
   , monomials((dim == 2) ? (1) : (k + 2))
   , n_pols(compute_n_pols(k))
```

Comment:
same here

Reply:
There's a concern with `PolynomialsBDM`. At the moment, it has its own function `degree()` which computes the degree as `polynomial_space.degree() - 1`. At the time that `TensorPolynomialsBase` is constructed, the quantity is unknown.

Should I leave that function as-is for now?

Reply:
Yes. Make a note about that to figure this out at a later time -- I usually use github issues for these notes so they don't get forgotten, and you should feel free to do the same :-)

Reply:
Okay #8591 is there to remind us about it. :smiley: